src/lib/storage/lz4_segment.cpp

```diff
@@ -488,8 +488,8 @@ std::optional<CompressedVectorType> LZ4Segment<T>::compressed_vector_type() cons
 template <>
 std::optional<CompressedVectorType> LZ4Segment<pmr_string>::compressed_vector_type() const {
   std::optional<CompressedVectorType> type;
-  if (_string_offsets) {
-    resolve_compressed_vector_type(*(*_string_offsets), [&](const auto& vector) { type = vector.type(); });
+  if (_string_offsets && *_string_offsets) {
+    return (*_string_offsets)->type();
```

Comment:
Why is `_string_offsets && !*_string_offsets` a valid case? We had a discussion about [optional<pointer_type>](https://github.com/hyrise/hyrise/pull/2070/files/cb7b16eb700d1bbef5746ee38bfc05f3c547bd00#r402193890) a while ago. To me, the bug fixed in this PR looks like a good example on why that ambiguity causes problems.

Reply:
Good point. Tbh, I haven't thought about to much and just cared about the segfault.

Personally, I'd remove the unique pointer as this vector is a member of the segment and shouldn't be shared or passed somewhere else. So ownership should be clear.
What do you think about `std::optional<std::vector__offsets>`?

Reply:
Ah, it's a base pointer. Doesn't work. 

I am wondering if I should delete the PR altogether, because I need to rework this part anyways. But who knows when this will be done. :(

What's your opinion on a unique ptr member which defaults to a nullptr. I don't like it, because I am afraid of pointers. They keep me awake at night.

Reply:
Yes. I haven't looked at the code around this, but judging from the PR, a unique_ptr that might be null would be cleaner.

Reply:
Changed it. I agree that it's better this way. Even though I still hate pointers. :D